### ISSUE TITLE
HiGHS hangs after/during presolve of LP

### ISSUE THREAD

**Issue by jannicklange** [ERGO-Code/HiGHS#1280]
Hi!

I am sometimes running into issues when trying to solve several instances of the same model.
For some of the instances, HiGHS never completes the presolve, and hangs.

This occurrs when calling it via the C# interface (through OPTANO.Modeling), but also (for some of the instances) when running the `highs.exe` (DL from [Julia](https://github.com/JuliaBinaryWrappers/HiGHSstatic_jll.jl/releases/tag/HiGHSstatic-v1.5.1%2B0)) on the command line.
The problem can be solved by disabling the `presolve`. However, this increases the runtime for successfull runs from a couple of minutes to more than 1 hour, which is not an option for my use case.

### Sample File(s)
I managed to create a relatively small LP that "reliably" crashes. The file is written by adding constraints via the HiGHS C# interface, and then calling the export function. Afterwards, I solve the model with the native highs.exe.

**NOTE**: When I randomize the order of constraints before adding them to the HiGHS model, the instance can be solved.
I did not start too many attempts, though (maybe ~15 minutes of "run test until failure", while each run takes about 1 minute).
My assumption would be that there will be more than 1 "bad" random order where the presolve will hang. However, it seems to be a relatively rare occurrence.

* [hanging_presolve.lp.txt](https://github.com/ERGO-Code/HiGHS/files/11380971/hanging_presolve.lp.txt)
* [highs.options.txt](https://github.com/ERGO-Code/HiGHS/files/11381005/highs.options.txt)
* PS Command: `.\highs .\hanging_presolve.lp --options_file=.\highs.options.txt`

### Log with enabled presolve:
```
Running HiGHS 1.5.1 [date: 1970-01-01, git hash: 93f1876e4]
Copyright (c) 2023 HiGHS under MIT licence terms
MIP  hanging_presolve has 184 rows; 192 cols; 506 nonzeros; 6 integer variables
Presolving model
55 rows, 66 cols, 161 nonzeros
14 rows, 24 cols, 43 nonzeros

(Process continues to run with 0% CPU utilization)
```

### Log without presolve:
```
Running HiGHS 1.5.1 [date: 1970-01-01, git hash: 93f1876e4]
Copyright (c) 2023 HiGHS under MIT licence terms
MIP  hanging_presolve has 184 rows; 192 cols; 506 nonzeros; 6 integer variables

Presolve is switched off

Solving MIP model with:
   184 rows
   192 cols (6 binary, 0 integer, 0 implied int., 186 continuous)
   506 nonzeros

        Nodes      |    B&B Tree     |            Objective Bounds              |  Dynamic Constraints |       Work      
     Proc. InQueue |  Leaves   Expl. | BestBound       BestSol              Gap |   Cuts   InLp Confl. | LpIters     Time

         0       0         0   0.00%   -inf            inf                  inf        0      0      0         0     0.1s
         0       0         0   0.00%   152000          inf                  inf        0      0      0         4     0.2s
 R       0       0         0   0.00%   152000          152000             0.00%        5      2      0         5     0.2s

Solving report
  Status            Optimal
  Primal bound      152000
  Dual bound        152000
  Gap               0% (tolerance: 0.01%)
  Solution status   feasible
                    152000 (objective)
                    0 (bound viol.)
                    0 (int. viol.)
                    0 (row viol.)
  Timing            0.24 (total)
                    0.01 (presolve)
                    0.00 (postsolve)
  Nodes             1
  LP iterations     5 (total)
                    0 (strong br.)
                    1 (separation)
                    0 (heuristics)

```

Please let me know, if you need me to provide any additional information.

Best Regards
Jannick

**Comment by jajhall** [comment]
I've verified that presolve hangs with https://github.com/ERGO-Code/HiGHS/tree/latest

I'm currently debugging the presolve, and if I run your instance it triggers an assert that I've just put into my development branch. I can't predict what might follow as a result of the error that it's trapped, but it might well explain what you've observed. 

I'll fix the bug I'm working on and see what happens with your instance

**Comment by jajhall** [comment]
Triggered an assert in `debug` due to illegal values in a presolve reduction. Unfortunate that it hangs forever in `release`. Nothing to do with the what I thought earlier. 

Fixed now, and soon to be pushed to [latest.](https://github.com/ERGO-Code/HiGHS/tree/latest)

**Comment by jajhall** [comment]
Closed by #1281
